src/services/powerManager.ts

```
@@ -0,0 +1,623 @@
+  private readonly defaultConfig: PowerOptimizationConfig = {
+    batteryThreshold: 20, // Switch to power saver at 20%
+    thermalThreshold: 65, // Throttle at 65°C
+    idleTimeoutMs: 300000, // 5 minutes idle timeout
+    aggressivePowerSaving: false,
+    adaptiveBrightness: true
+  };
+
+  constructor(config?: Partial<PowerOptimizationConfig>) {
+    if (PowerManager.instance) {
+      return PowerManager.instance;
+    }
```

Biome (2.1.2)

[error] The constructor should not return a value.

The constructor is here:

Returning a value from a constructor may confuse users of the class.

(lint/correctness/noConstructorReturn)